lib/buffer.js

```diff
@@ -637,21 +638,18 @@ Buffer.prototype.toString = function toString(encoding, start, end) {
   }
 
   const len = this.length;
-  if (len === 0)
-    return '';
 
-  if (!start || start < 0)
+  if (start <= 0)
     start = 0;
   else if (start >= len)
     return '';
+  else
+    start |= 0;
```

Comment:
Uh … I think moving this into the `else` makes behaviour different for e.g. numeric strings?